pkg/user/tx_client.go

```diff
@@ -546,8 +603,54 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 			if err := client.signer.SetSequence(signer, sequence); err != nil {
 				return nil, fmt.Errorf("setting sequence: %w", err)
 			}
+
+			// When a previous transaction is rejected due to TTL, all subsequent pending transactions will be rejected for sequence mismatch. The network expects lower sequence since previous transactions were not committed.
+			if apperrors.IsNonceMismatchCode(resp.ExecutionCode) {
+				networkSequence, err := client.queryNetworkSequence(ctx, signer)
+				if err != nil {
+					return nil, fmt.Errorf("failed to query network sequence: %w", err)
+				}
+
+				// If network sequence is greater than cached sequence, a tx was confirmed
+				// and the client is not aware of it.
+				if networkSequence > sequence {
+					return nil, fmt.Errorf("tx with hash %s was rejected by the node with log: %s", txHash, resp.Error)
+				}
+
+				// Set sequence to the network sequence to resign and resubmit the transaction
+				if err := client.signer.SetSequence(signer, networkSequence); err != nil {
+					return nil, fmt.Errorf("setting sequence: %w", err)
+				}
```

Comment:
This won't work, if a transaction is rejected and there are several subsequent transactions, each of those subsequent ones will all get set with the same network sequence and thus only the first thereafter will be accepted, the rest will be immediately rejected in CheckTx